src/python_testing/TC_SU_2_2.py

```diff
@@ -0,0 +1,1004 @@
+                     "which is 2 minutes (120 seconds) from the last QueryImage command."),
+            TestStep(4, "DUT sends a QueryImage command to the TH/OTA-P. TH/OTA-P sends a QueryImageResponse back to DUT. "
+                     "QueryStatus is set to Busy, Set DelayedActionTime to 3 minutes. On the subsequent QueryImage command, "
+                     "TH/OTA-P sends a QueryImageResponse back to DUT. QueryStatus is set to 'UpdateAvailable'.",
+                     "Verify that the DUT waits for at least the time mentioned in the DelayedActionTime (3 minutes) before issuing another QueryImage command to the TH/OTA-P."
+                     "Verify that there is a transfer of the software image after the second QueryImageResponse with UpdateAvailable status from the TH/OTA-P to the DUT."),
+            TestStep(5, "DUT sends a QueryImage command to the TH/OTA-P. TH/OTA-P sends a QueryImageResponse back to DUT. QueryStatus is set to 'UpdateAvailable'",
+                     "ImageURI should have the https url from where the image can be downloaded.",
+                     "Verify that the DUT queries the https url and downloads the software image."),
+            TestStep(6, "DUT sends a QueryImage command to the TH/OTA-P. TH/OTA-P sends a QueryImageResponse back to DUT. QueryStatus is set to 'UpdateAvailable'",
+                     "Software Version should be set to the same or an older (numerically lower) version.",
+                     "Verify that the DUT does not start transferring the software image."),
+            TestStep(7, "DUT sends a QueryImage command to the TH/OTA-P. TH/OTA-P sends a QueryImageResponse back to DUT. "
+                     "QueryStatus is set to 'UpdateAvailable', ImageURI field contains an invalid BDX ImageURI.",
+                     "Verify that the DUT does not start transferring the software image.")
+        ]
+        return steps
+
+    @async_test_body
+    async def teardown_test(self):
+        await self.clear_ota_providers(self.default_controller, self.dut_node_id)
+        self.terminate_provider()
+        super().teardown_test()
+
+    @async_test_body
+    async def test_TC_SU_2_2(self):
+        self.LOG_FILE_PATH = "provider.log"
+        self.KVS_PATH = "/tmp/chip_kvs_provider"
+        self.provider_app_path = self.user_params.get('provider_app_path', None)
+        self.ota_image = self.user_params.get('ota_image')
+
+        self.step(0)
+        # Controller has already commissioned the requestor
+
+        # Prerequisite #1.0 - Requestor (DUT) info
+        controller = self.default_controller
+        fabric_id = controller.fabricId
+        requestor_node_id = self.dut_node_id
+
+        # Prerequisite #1.0 - Provider info
+        provider_node_id = 1
+        provider_discriminator = 1111
+        provider_setupPinCode = 20202021
+        provider_port = self.user_params.get('ota_provider_port', 5541)
+
+        self.provider_data = {
```

Comment:
please use a dataclass rather than a string dict.

Reply:
Actually, why do you need this packed up at all? They're unpacked at the call site. Why not just use the variables directly?